In [2]:
import pandas as pd
import matplotlib as plt

df = pd.read_csv('permisos_construccion_2.csv')
print(df.head())

C:\Users\gaspa\AppData\Local\Temp\ipykernel_12812\792157118.py:4: DtypeWarning: Columns (22,32,38) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('permisos_construccion_2.csv')


  Permit Number  Permit Type  Permit Type Definition Permit Creation Date  \
0       M788927            8  otc alterations permit           05/23/2017   
1  201305318356            8  otc alterations permit           05/31/2013   
2  201705106205            8  otc alterations permit           05/10/2017   
3  201410279983            8  otc alterations permit           10/27/2014   
4  201310280388            8  otc alterations permit           10/28/2013   

  Block   Lot  Street Number Street Number Suffix Street Name Street Suffix  \
0  0215   001           1333                  NaN       jOnEs            St   
1  1810  017A           1483                  NaN        43rD            Av   
2  5700   027            431                  NaN    pReNtIsS            St   
3  0661   005           2020                  NaN        bUsH            St   
4  3642  051A            871                  NaN        cApP            St   

   ...  Existing Construction Type Existing Construction Type 

In [3]:
# a. Cant de Filas y Columnas
print("Cant Filas", df.shape[0])
print("Cant Columnas", df.shape[1])

Cant Filas 198910
Cant Columnas 43


In [4]:
# b. Primeras 5 filas
print("Hay filas duplicadas?", df.head())

Hay filas duplicadas?   Permit Number  Permit Type  Permit Type Definition Permit Creation Date  \
0       M788927            8  otc alterations permit           05/23/2017   
1  201305318356            8  otc alterations permit           05/31/2013   
2  201705106205            8  otc alterations permit           05/10/2017   
3  201410279983            8  otc alterations permit           10/27/2014   
4  201310280388            8  otc alterations permit           10/28/2013   

  Block   Lot  Street Number Street Number Suffix Street Name Street Suffix  \
0  0215   001           1333                  NaN       jOnEs            St   
1  1810  017A           1483                  NaN        43rD            Av   
2  5700   027            431                  NaN    pReNtIsS            St   
3  0661   005           2020                  NaN        bUsH            St   
4  3642  051A            871                  NaN        cApP            St   

   ...  Existing Construction Type Exist

In [60]:
# c. Datos Faltantes y Duplicados

# # Faltantes
missing_values = df.isnull().sum()

for columna, num_missing in missing_values.items():
    if num_missing > 0:
        print(f"{columna} faltan {num_missing}")

# Duplicados
num_duplicados = df.duplicated().sum()
print(f"Número de filas duplicadas: {num_duplicados}")



Street Number Suffix faltan 196694
Street Suffix faltan 2768
Unit faltan 169430
Unit Suffix faltan 196949
Description faltan 290
Issued Date faltan 14942
Completed Date faltan 101715
First Construction Document Date faltan 14948
Structural Notification faltan 191988
Number of Existing Stories faltan 42788
Number of Proposed Stories faltan 42871
Voluntary Soft-Story Retrofit faltan 198875
Fire Only Permit faltan 180082
Permit Expiration Date faltan 51884
Estimated Cost faltan 38068
Revised Cost faltan 6066
Existing Use faltan 41117
Existing Units faltan 51543
Proposed Use faltan 42441
Proposed Units faltan 50915
Plansets faltan 37311
TIDF Compliance faltan 198908
Existing Construction Type faltan 43369
Existing Construction Type Description faltan 43370
Proposed Construction Type faltan 43165
Proposed Construction Type Description faltan 43165
Site Permit faltan 193550
Supervisor District faltan 1719
Neighborhoods - Analysis Boundaries faltan 1725
Zipcode faltan 1716
Location faltan 170

In [61]:
# d. Causas de Datos Faltantes



 Street Number Suffix: tiene 196694 datos faltanes, probablemente porque muchas direcciones no tienen un sufijo en su numero
Street Suffix tiene 2768 datos faltantes, probablemente son errores de ingreso de datos ya que todas las casas tienen sufijo de calle
Unit tiene 169430 datos faltantes, 
Unit Suffix tiene 196949 datos faltanes, igual que Street Number Suffix
Description: tiene 290 datos faltantes, 
Issued Date                                 14942
Completed Date                             101715
First Construction Document Date            14948
Structural Notification                    191988
Number of Existing Stories                  42788
Number of Proposed Stories                  42871
Voluntary Soft-Story Retrofit              198875
Fire Only Permit                           180082
Permit Expiration Date                      51884
Estimated Cost                              38068
Revised Cost                                 6066
Existing Use                                41117
Existing Units                              51543
Proposed Use                                42441
Proposed Units                              50915
Plansets                                    37311
TIDF Compliance                            198908
Existing Construction Type                  43369
Existing Construction Type Description      43370
Proposed Construction Type                  43165
...
Neighborhoods - Analysis Boundaries                               1725
Zipcode                               1716
Location                               1700


In [62]:

# Evaluar los valores únicos para cada columna
for column in df.columns:
    unique_values = df[column].unique()
    print(f"Valores únicos de {column}: {unique_values}")
    print(f"Cantidad de valores únicos en {column}: {len(unique_values)}\n")

Valores únicos de Permit Number: ['M788927' '201305318356' '201705106205' ... '201607293741' '201701066691'
 '201604255599']
Cantidad de valores únicos en Permit Number: 181495

Valores únicos de Permit Type: [8 2 4 3 6 1 7 5]
Cantidad de valores únicos en Permit Type: 8

Valores únicos de Permit Type Definition: ['otc alterations permit' 'new construction wood frame' 'sign - erect'
 'additions alterations or repairs' 'demolitions' 'new construction'
 'wall or painted sign' 'grade or quarry or fill or excavate'
 ' otc alterations permit ' ' new construction '
 'otc alterations permit #' 'new construction #'
 ' additions alterations or repairs ']
Cantidad de valores únicos en Permit Type Definition: 13

Valores únicos de Permit Creation Date: ['05/23/2017' '05/31/2013' '05/10/2017' ... '2017-02-10' '2016-11-14'
 '2013-03-06']
Cantidad de valores únicos en Permit Creation Date: 1296

Valores únicos de Block: ['0215' '1810' '5700' ... '4976' '6125' '6757']
Cantidad de valores únicos en Bl